Turf/Turf.swift

```diff
@@ -285,3 +288,67 @@ public struct Polyline {
         return closestCoordinate
     }
 }
+
+public struct Polygon {
+    var polygonCoordinates: [[CLLocationCoordinate2D]]
+    
+    var polygonArea: Double {
+        var area:Double = 0
+   
+        if (!polygonCoordinates.isEmpty && polygonCoordinates.count > 0) {
+            
+            area += abs(ringArea(polygonCoordinates[0]))
+
+            for coordinate in polygonCoordinates.suffix(from: 1) {
+                area -= abs(ringArea(coordinate))
+            }
+        }
+        return area
+    }
+    
+    /**
+     * Calculate the approximate area of the polygon were it projected onto the earth.
+     * Note that this area will be positive if ring is oriented clockwise, otherwise it will be negative.
+     *
+     * Reference:
+     * Robert. G. Chamberlain and William H. Duquette, "Some Algorithms for Polygons on a Sphere", JPL Publication 07-03, Jet Propulsion
+     * Laboratory, Pasadena, CA, June 2007 http://trs-new.jpl.nasa.gov/dspace/handle/2014/40409
+     *
+     */
+    private func ringArea(_ coordinates: [CLLocationCoordinate2D]) -> Double {
+        var p1: CLLocationCoordinate2D
+        var p2: CLLocationCoordinate2D
+        var p3: CLLocationCoordinate2D
+        var lowerIndex: Int
+        var middleIndex: Int
+        var upperIndex: Int
```

Comment:
Use a tuple consisting of three coordinates to ensure that all three are set simultaneously:

```swift
let controlPoints: (CLLocationCoordinate2D, CLLocationCoordinate2D, CLLocationCoordinate2D)
if index == coordinatesCount - 2 {
    controlPoints = (coordinates[coordinatesCount - 2], coordinates[coordinatesCount - 1], coordinates[0])
}
// …
```